### PR TITLE
[RavenDB-10711] 

### PR DESCRIPTION
**Pull request by redknightlois**
Clean history of intrinsics based Compare, Diff and Search algorithms.

### CODE REVIEW

File: codealike.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"projectId":"6944aade-3195-477c-8aed-9ebe2902fc57","projectName":"RavenDB","neverTrack":false}
```
Comment (ayende) [inline codealike.json:1]: Probably this should be removed, no?

File: src/Sparrow/Utils/DiffPages.cs [hunk 2]
```diff
@@ -20,90 +21,237 @@ public unsafe class DiffPages
...
+            outputPtr[0] = startIdx * sizeof(ulong);
+            outputPtr[1] = runLengthInBytes;
+            outputSize += sizeof(long) * 2;
+
+            Unsafe.CopyBlock(Output + outputSize, (srcPtr + destOffset) + startIdx, (uint)runLengthInBytes);
+
+            OutputSize = outputSize + runLengthInBytes;
+        }
+
+        //public void ComputeDiff(void* originalBuffer, void* modifiedBuffer, int size)
```
Comment (ayende) [inline src/Sparrow/Utils/DiffPages.cs]: Can be removed?

File: codealike.json (hunk 1 above)
Comment (redknightlois) [reply codealike.json:1]: I use Codealike to measure myself, that is the correlation token (it used to exist at the solution level)  after many requests we moved it to a standalone file. :)

File: src/Sparrow/Utils/DiffPages.cs (hunk 2 above)
Comment (redknightlois) [reply src/Sparrow/Utils/DiffPages.cs]: Mhhh my bad, will not have to modify that (that was used for testing purposes). Removed and pushed already.

**Comment by redknightlois** [comment]
This commit is the output of the investigation, not the actual production code that would end up being included in v4.1 ... I need fallbacks when not supported, to be the actual code, etc...

**Comment by redknightlois** [comment]
That is the reason why this commit only modifies files in the bench subdirectory.

**Comment by ayende** [comment]
This actually changes these as well:

```
src/Sparrow/Binary/Bits.cs                         |   24 +-
 src/Sparrow/Compression/LZ4.cs                     |    2 +-
 src/Sparrow/Memory.cs                              |  190 +-
```

**Pull request merged**